bitcoin/src/address.rs

```diff
@@ -554,8 +564,54 @@ impl<'a> fmt::Display for AddressEncoding<'a> {
     }
 }
 
+/// Marker of status of address's network validation.
+pub trait NetworkValidation {}
+
+/// Marker that address's network has been successfully validated.
+#[derive(PartialEq)]
+pub enum NetworkValid {}
```

Comment:
Annoyingly, you actually have to derive every single trait that we derive on `Address`, otherwise the derives on `Address` won't do anything.

It just happens that `PartialEq` is the only one we test for.

I'm not sure if there's a good way to check for this :/

Reply:
Added all the derives from `Address` in 0e448ae3398cd4486855693f524c7e0e40f7f902.